services/121-service/src/payments/services/payments-creation.service.ts

```diff
@@ -0,0 +1,313 @@
+    query: PaginateQuery;
+    dryRun: boolean;
+    note?: string;
+  }): Promise<BulkActionResultPaymentDto> {
+    if (!dryRun) {
+      await this.paymentsProgressHelperService.checkPaymentInProgressAndThrow(
+        programId,
+      );
+    }
+
+    // TODO: REFACTOR: Move what happens in setQueryPropertiesBulkAction into this function, and call a refactored version of getBulkActionResult/getPaymentBaseQuery (create solution design first)
+    const paginateQuery =
+      this.registrationsBulkService.setQueryPropertiesBulkAction({
+        query,
+        includePaymentAttributes: true,
+      });
+
+    // Fill bulkActionResultDto with meta data of the payment being done
+    const bulkActionResultDto =
+      await this.registrationsBulkService.getBulkActionResult(
+        paginateQuery,
+        programId,
+        this.getPaymentBaseQuery(), // We need to create a separate queryBuilder object twice or it will be modified twice
+      );
+
+    // If amount is not defined do not calculate the totalMultiplierSum
+    // This happens when you call the endpoint with dryRun=true
+    // Calling with dryrun is true happens in the pa table when you try to do a payment to decide which registrations are selectable
+    if (!transferValue) {
+      return {
+        ...bulkActionResultDto,
+        sumPaymentAmountMultiplier: 0,
+        programFspConfigurationNames: [],
+      };
+    }
+
+    // Get array of RegistrationViewEntity objects to be paid
+    const registrationsForPayment =
+      await this.getRegistrationsForPaymentChunked(programId, paginateQuery);
+
+    // Calculate the totalMultiplierSum and create an array with all FSPs for this payment
+    // Get the sum of the paymentAmountMultiplier of all registrations to calculate the total amount of money to be paid in frontend
+    let totalMultiplierSum = 0;
+    // This loop is pretty fast: with 131k registrations it takes ~38ms
+
+    for (const registration of registrationsForPayment) {
+      totalMultiplierSum =
+        totalMultiplierSum + registration.paymentAmountMultiplier;
+    }
+
+    // Get unique programFspConfigurationNames in payment
+    // Getting unique programFspConfigurationNames is relatively: with 131k registrations it takes ~36ms locally
+    const programFspConfigurationNames = Array.from(
+      new Set(
+        registrationsForPayment.map(
+          (registration) => registration.programFspConfigurationName,
+        ),
+      ),
+    );
+
+    // Fill bulkActionResultPaymentDto with bulkActionResultDto and additional payment specific data
+    const bulkActionResultPaymentDto: BulkActionResultPaymentDto = {
+      ...bulkActionResultDto,
+      sumPaymentAmountMultiplier: totalMultiplierSum,
+      programFspConfigurationNames,
+    };
+
+    // Create an array of referenceIds to be paid
+    const referenceIds = registrationsForPayment.map(
+      (registration) => registration.referenceId,
+    );
+
+    if (!dryRun && referenceIds.length > 0) {
```

Comment:
I think this pattern is better. See pseudo code: 
```suggestion
    if (!dryRun) {
    	if (referenceIds.length < 1) {
    	  throw new HttpException('No registrations')
    	}
    	
```
Maybe out of scope?